v1.0-beta doesn't recognize some of my iterators

Some of my views are not accepted by v1.0 beta anymore, I think have tracked the issue to your new concept check for InputIterator requiring `DerivedFrom<detail::iter_concept_t<I>, ranges::input_iterator_tag>`.

This call is ambiguous for iterators that inherit from gnu stl iterators which is the case for my affected views:
```
/home/mi/h4nn3s/devel/seqan3/submodules/range-v3/include/range/v3/iterator/concepts.hpp: In substitution of 'template<class I> using iter_concept_t = decltype (ranges::detail::iter_concept_(declval<I>(), ranges::detail::priority_tag<3>{})) [with I = seqan3::detail::view_take_line<ranges::ref_view<std::__cxx11::basic_string<char> >, false>::iterator_type<ranges::ref_view<std::__cxx11::basic_string<char> > >]':
/home/mi/h4nn3s/devel/seqan3/test/unit/range/view/view_take_line_test.cpp:103:76:   required from here
/home/mi/h4nn3s/devel/seqan3/submodules/range-v3/include/range/v3/iterator/concepts.hpp:102:54: error: call of overloaded 'iter_concept_(seqan3::detail::view_take_line<ranges::ref_view<std::__cxx11::basic_string<char> >, false>::iterator_type<ranges::ref_view<std::__cxx11::basic_string<char> > >, ranges::detail::priority_tag<3>)' is ambiguous
         using iter_concept_t = decltype(iter_concept_(std::declval<I>(), priority_tag<3>{}));
                                         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mi/h4nn3s/devel/seqan3/submodules/range-v3/include/range/v3/iterator/concepts.hpp:49:14: note: candidate: ranges::contiguous_iterator_tag ranges::detail::iter_concept_(__gnu_cxx::__normal_iterator<T*, Seq>, ranges::detail::priority_tag<3>) [with T = char; Seq = std::__cxx11::basic_string<char>]
         auto iter_concept_(__gnu_cxx::__normal_iterator<T *, Seq>, priority_tag<3>) ->
              ^~~~~~~~~~~~~
/home/mi/h4nn3s/devel/seqan3/submodules/range-v3/include/range/v3/iterator/concepts.hpp:94:14: note: candidate: typename ranges::detail::if_then<is_std_iterator_traits_specialized_<I> >::apply<std::iterator_traits<_Iter>, I>::iterator_category ranges::detail::iter_concept_(I, ranges::detail::priority_tag<1>) [with I = seqan3::detail::view_take_line<ranges::ref_view<std::__cxx11::basic_string<char> >, false>::iterator_type<ranges::ref_view<std::__cxx11::basic_string<char> > >; typename ranges::detail::if_then<is_std_iterator_traits_specialized_<I> >::apply<std::iterator_traits<_Iter>, I>::iterator_category = std::random_access_iterator_tag]
         auto iter_concept_(I, priority_tag<1>) -> typename iter_traits_t<I>::iterator_category;
              ^~~~~~~~~~~~~
/home/mi/h4nn3s/devel/seqan3/submodules/range-v3/include/range/v3/iterator/concepts.hpp:96:14: note: candidate: ranges::detail::enable_if_t<(! is_std_iterator_traits_specialized_<I>), std::random_access_iterator_tag> ranges::detail::iter_concept_(I, ranges::detail::priority_tag<0>) [with I = seqan3::detail::view_take_line<ranges::ref_view<std::__cxx11::basic_string<char> >, false>::iterator_type<ranges::ref_view<std::__cxx11::basic_string<char> > >; ranges::detail::enable_if_t<(! is_std_iterator_traits_specialized_<I>), std::random_access_iterator_tag> = std::random_access_iterator_tag]
         auto iter_concept_(I, priority_tag<0>) ->
              ^~~~~~~~~~~~~
/home/mi/h4nn3s/devel/seqan3/test/unit/range/view/view_take_line_test.cpp: In member function 'virtual void view_take_line_unix_eol_Test::TestBody()':
/home/mi/h4nn3s/devel/seqan3/test/unit/range/view/view_take_line_test.cpp:103:26: error: template argument 1 is invalid
     std::cout << detail::get_display_name_v<ranges::detail::iter_concept_t<decltype(begin(v))>>.string() << '\n';
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gmake[3]: *** [range/view/CMakeFiles/view_take_line_test.dir/build.make:63: range/view/CMakeFiles/view_take_line_test.dir/view_take_line_test.cpp.o] Fehler 1
gmake[2]: *** [CMakeFiles/Makefile2:5333: range/view/CMakeFiles/view_take_line_test.dir/all] Fehler 2
gmake[1]: *** [CMakeFiles/Makefile2:5345: range/view/CMakeFiles/view_take_line_test.dir/rule] Fehler 2
gmake: *** [Makefile:1650: view_take_line_test] Fehler 2
```

I assume you are doing this hack to whitelist some STL random access iterators as contiguous iterators. But since you are relying on conversion for the priority, too, this becomes ambiguous. This is a nasty problem to detect, because you don't immediately get the above error, instead the concept check just silently fails.

--- a/include/range/v3/iterator/concepts.hpp
+++ b/include/range/v3/iterator/concepts.hpp
@@ -40,19 +40,22 @@ namespace ranges
             if_then_t<is_std_iterator_traits_specialized_<I>, std::iterator_traits<I>, I>;
 
 #if defined(_GLIBCXX_DEBUG)
-        template<typename T, typename Seq>
+        template<typename I, typename T, typename Seq>
         auto iter_concept_(__gnu_debug::_Safe_iterator<T *, Seq>, priority_tag<3>) ->
-            ranges::contiguous_iterator_tag;
+            CPP_ret(ranges::contiguous_iterator_tag)(
+                requires Same<I, __gnu_debug::_Safe_iterator<T *, Seq>>);
 #endif
 #if defined(__GLIBCXX__)
-        template<typename T, typename Seq>
+        template<typename I, typename T, typename Seq>
         auto iter_concept_(__gnu_cxx::__normal_iterator<T *, Seq>, priority_tag<3>) ->
-            ranges::contiguous_iterator_tag;
+            CPP_ret(ranges::contiguous_iterator_tag)(
+                requires Same<I, __gnu_cxx::__normal_iterator<T *, Seq>>);
 #endif
 #if defined(_LIBCPP_VERSION)
-        template<typename T>
+        template<typename I, typename T>
         auto iter_concept_(std::__wrap_iter<T *>, priority_tag<3>) ->
-            ranges::contiguous_iterator_tag;
+            CPP_ret(ranges::contiguous_iterator_tag)(
+                requires Same<I, std::__wrap_iter<T *>>);
 #endif
 #if defined(_MSVC_STL_VERSION)
         template<typename I>
@@ -86,8 +89,10 @@ namespace ranges
             CPP_ret(ranges::contiguous_iterator_tag)(
                 requires Same<I, class I::_String_view_iterator>);
 #endif
-        template<typename T>
-        auto iter_concept_(T *, priority_tag<3>) -> ranges::contiguous_iterator_tag;
+        template<typename I, typename T>
+        auto iter_concept_(T *, priority_tag<3>) ->
+            CPP_ret(ranges::contiguous_iterator_tag)(
+                requires Same<I, T *>);
         template<typename I>
         auto iter_concept_(I, priority_tag<2>) -> typename iter_traits_t<I>::iterator_concept;
         template<typename I>
@@ -99,7 +104,7 @@ namespace ranges
                 ranges::random_access_iterator_tag>;
 
         template<typename I>
-        using iter_concept_t = decltype(iter_concept_(std::declval<I>(), priority_tag<3>{}));
+        using iter_concept_t = decltype(iter_concept_<I>(std::declval<I>(), priority_tag<3>{}));
     }
     /// \endcond
 

